packages/aws-cdk-lib/aws-events-targets/lib/aws-api.ts

```diff
@@ -81,12 +81,9 @@ export class AwsApi implements events.IRuleTarget {
    * result from an EventBridge event.
    */
   public bind(rule: events.IRule, id?: string): events.RuleTargetConfig {
-    const handler = new lambda.SingletonFunction(rule as events.Rule, `${rule.node.id}${id}Handler`, {
-      code: lambda.Code.fromAsset(path.join(__dirname, '..', '..', 'custom-resource-handlers', 'dist', 'aws-events-targets', 'aws-api-handler')),
-      runtime: lambda.Runtime.NODEJS_18_X,
+    const handler = new AwsApiProvider(rule as events.Rule, `${rule.node.id}${id}Handler`, {
```

Comment:
This is not actually a custom resource though (so it's also not really a "provider"). I'm not opposed to including it, but aren't we stretching naming conventions here a bit?